hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -70,34 +73,59 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
     )
     .getOrElse(new Properties())
 
+  private val whitelistedPayloadClasses: Set[String] = Seq(
+    classOf[OverwriteWithLatestAvroPayload]
```

Comment:
The problem with `DefaultHoodieRecordPayload` that it's also requiring `PAYLOAD_EVENT_TIME_FIELD_PROP_KEY` field. In the future when we migrate to unified Merge API we would allow particular configured Merge Engine to expose which fields are required for it and then make Relation load those. For now, i opted in to keep things simple here until we reach that point. WDYT?

Reply:
sounds ok to make it work with OverwriteWithLatestAvroPayload which is the current default, covering most cases